### PR TITLE
[BugFix] Removed a lot of context and replaced with the theme

### LINKED ISSUE #424: Wrong Theme of the error pop-up of Location Service
**Describe the bug**
The location error pop-up when you press the location button inside the search bar is with the wrong theme

**To Reproduce**
Steps to reproduce the behaviour:
1. Select Your Location Button on Search Screen

**Expected behaviour**
Should look exactly like the one if you click on the GPS icon.

**Screenshots**

Introduces by #420

### PR DESCRIPTION
**Pull request by md-weber**
The theme needed to be passed down because the search field did not get the correct Theme data from its context. 

Resolves #424

### CODE REVIEW

**Review by GustavoTCh — COMMENTED** [review]
It's a good solution.

File: lib/location/location_search_delegate.dart [hunk 1]
```diff
@@ -65,7 +65,10 @@ class LocationSearchDelegate extends SearchDelegate<TrufiLocation> {
 
   @override
   Widget buildSuggestions(BuildContext context) {
+    final theme = Theme.of(context);
+
```
Comment (SamuelRioTz) [inline lib/location/location_search_delegate.dart:69]: why the theme is passed as parameter, I thought it should be part of the context? maybe this issues is more related with dialog build context issues

Comment (GustavoTCh) [reply lib/location/location_search_delegate.dart:69]: https://github.com/trufi-association/trufi-core/blob/73b00220cbcf21cabdd7aff8f9e9855d2adf3a9f/lib/location/location_search_delegate.dart#L31 

The LoactionSearchDelegate defines the theme, background = white.

Comment (SamuelRioTz) [reply lib/location/location_search_delegate.dart:69]: so I think passing theme as parameter is not the good way, maybe we can override the LocationSearchDelegate background after used

Comment (md-weber) [reply lib/location/location_search_delegate.dart:69]: Yeah, I am also not happy with the solution. I tried it lower in the widget tree but as soon I get one level down it directly get overwritten from somewhere. The problem is that the SearchDelegate overwrites the Theme of the context and uses its own. 

Currently, I did not have a solution for that and actually did not wanted to invest so much time. But I can investigate where the Problem happens.

**Review by GustavoTCh — APPROVED** [review]

**Comment by md-weber** [comment]
I would close this PR in favor for the discussed solution with @SamuelRioTz

**Pull request closed without merging**